lz4/lz4.go

```diff
@@ -31,23 +48,34 @@ func (c CompressionCodec) Code() int8 {
 func (c CompressionCodec) Encode(src []byte) ([]byte, error) {
 	buf := bytes.Buffer{}
 	buf.Grow(len(src)) // guess a size to avoid repeat allocations.
-	writer := lz4.NewWriter(&buf)
+	writer := writerPool.Get().(*lz4.Writer)
+	writer.Reset(&buf)
 
 	_, err := writer.Write(src)
 	if err != nil {
+		// don't return writer to pool on error.
 		return nil, err
 	}
 
-	err = writer.Close()
+	err = writer.Flush()
```

Comment:
Is calling Reset after Close not working? Or is Close not writing anything to the underlying stream?

Reply:
Good catch.  Using `Flush` instead of `Close` here is what lead to the test failure.